genvm/vm.go

```diff
@@ -231,6 +231,12 @@ func (v *VM) Apply(lctx ApplyContext, txs []types.Transaction, blockRewards []ty
 		if err := rewards.Add(tx, &reward); err != nil {
 			return nil, nil, fmt.Errorf("%w: %s", core.ErrInternal, err.Error())
 		}
+		events.ReportRewardReceived(events.Reward{
+			Layer:       reward.Layer,
+			Total:       reward.TotalReward,
+			LayerReward: reward.LayerReward,
+			Coinbase:    reward.Coinbase,
+		})
```

Comment:
have you checked if rewards are read from the database when event is received? that was majority of streams were "designed"
the problem is if event is received before tx is committed, rewards won't be available in the database

Reply:
I changed it to report rewards events **after** transactions are saved in DB.